rpc/src/lib.rs

```diff
@@ -228,8 +228,23 @@ impl<B, C, SC, P, CT, BE> EthApiT for EthApi<B, C, SC, P, CT, BE> where
 		unimplemented!("proof");
 	}
 
-	fn storage_at(&self, _: H160, _: U256, _: Option<BlockNumber>) -> BoxFuture<H256> {
-		unimplemented!("storage_at");
+	fn storage_at(&self, address: H160, index: U256, number: Option<BlockNumber>) -> Result<H256> {
+		if let Some(number) = number {
+			if number != BlockNumber::Latest {
+				unimplemented!("fetch storage for past blocks is not yet supported");
```

Comment:
Just a note that we can support this easily -- simply changing the `BlockId::Hash(header.hash())` below to use the number requested. Maybe a separate PR though.

Reply:
Agreed, probably a separate PR.